File: ios/Sources/VaultSync/Presentation/ViewModels/SettingsViewModel.swift
import SwiftUI
import Combine

public struct SettingsState: Sendable {
    public var cipherProtocol: String = "AES-256-GCM"
    public var keyStorage: String = "Apple Keychain (Secure Enclave)"
    public var isWifiOnly: Bool = true
    public var isBackgroundSyncEnabled: Bool = true
    public var totalStorageBytes: Int64 = 0
    public var documentCount: Int = 0

    public init() {}
}

@MainActor
public final class SettingsViewModel: ObservableObject {
    @Published public var state = SettingsState()

    private let documentRepository: any DocumentRepository

    public init(documentRepository: any DocumentRepository) {
        self.documentRepository = documentRepository
        refreshStorage()
    }

    public func refreshStorage() {
        Task {
            if let docs = try? await documentRepository.getDocuments() {
                state.totalStorageBytes = docs.reduce(0) { $0 + $1.size }
                state.documentCount = docs.count
            }
        }
    }

    public func toggleWifiOnly(_ enabled: Bool) {
        state.isWifiOnly = enabled
    }

    public func toggleBackgroundSync(_ enabled: Bool) {
        state.isBackgroundSyncEnabled = enabled
    }
}

File: android/app/src/main/java/com/vaultsync/ui/viewmodel/SettingsViewModel.kt
package com.vaultsync.ui.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.vaultsync.domain.repository.DocumentRepository
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch

data class SettingsState(
    val encryptionAlgorithm: String = "AES-256-GCM",
    val keyStoreProvider: String = "AndroidKeyStore (Hardware TEE)",
    val isWifiOnly: Boolean = true,
    val isBackgroundSyncEnabled: Boolean = true,
    val totalStorageUsedBytes: Long = 0L,
    val documentCount: Int = 0
)

class SettingsViewModel(
    private val documentRepository: DocumentRepository
) : ViewModel() {

    private val _state = MutableStateFlow(SettingsState())
    val state: StateFlow<SettingsState> = _state.asStateFlow()

    init {
        loadSettings()
    }

    fun loadSettings() {
        viewModelScope.launch {
            val docs = documentRepository.getDocuments()
            val totalBytes = docs.sumOf { it.size }
            _state.value = _state.value.copy(
                totalStorageUsedBytes = totalBytes,
                documentCount = docs.size
            )
        }
    }

    fun toggleWifiOnly(enabled: Boolean) {
        _state.value = _state.value.copy(isWifiOnly = enabled)
    }

    fun toggleBackgroundSync(enabled: Boolean) {
        _state.value = _state.value.copy(isBackgroundSyncEnabled = enabled)
    }
}
